batch/snippets/src/test/java/BatchTemplateIT.java

```diff
@@ -0,0 +1,210 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth.assertWithMessage;
+
+import com.google.cloud.compute.v1.AccessConfig;
+import com.google.cloud.compute.v1.AccessConfig.NetworkTier;
+import com.google.cloud.compute.v1.AttachedDisk;
+import com.google.cloud.compute.v1.AttachedDiskInitializeParams;
+import com.google.cloud.compute.v1.DeleteInstanceTemplateRequest;
+import com.google.cloud.compute.v1.InsertInstanceTemplateRequest;
+import com.google.cloud.compute.v1.InstanceProperties;
+import com.google.cloud.compute.v1.InstanceTemplate;
+import com.google.cloud.compute.v1.InstanceTemplatesClient;
+import com.google.cloud.compute.v1.NetworkInterface;
+import com.google.cloud.compute.v1.Operation;
+import com.google.cloud.compute.v1.Scheduling;
+import com.google.cloud.compute.v1.Scheduling.OnHostMaintenance;
+import com.google.cloud.compute.v1.Scheduling.ProvisioningModel;
+import com.google.cloud.compute.v1.ServiceAccount;
+import com.google.cloud.resourcemanager.v3.ProjectsClient;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Arrays;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class BatchTemplateIT {
+
+  private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static final String REGION = "europe-north1";
+  private static String PROJECT_NUMBER;
+  private static String SCRIPT_JOB_NAME;
+  private static InstanceTemplate INSTANCE_TEMPLATE;
+
+  private ByteArrayOutputStream stdOut;
+
+  // Check if the required environment variables are set.
+  public static void requireEnvVar(String envVarName) {
+    assertWithMessage(String.format("Missing environment variable '%s' ", envVarName))
+        .that(System.getenv(envVarName))
+        .isNotEmpty();
+  }
+
+  @BeforeClass
+  public static void setUp()
+      throws IOException, InterruptedException, ExecutionException, TimeoutException {
+    final PrintStream out = System.out;
+    ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
```

Comment:
Are there any resources that might be left over if one of your tests fail?  
- Should old >24h ones be cleaned up here?

Reply:
Thanks Les for pointing it out. 
I've added code to delete stale and hanging resources. PTAL.